contractcourt/chain_arbitrator.go

```diff
@@ -603,21 +604,60 @@ func (c *ChainArbitrator) Start() error {
 		close(watcherErrs)
 	}()
 
+	// stopAndLog is a helper function which shuts down the chain arb and
+	// logs errors if they occur.
+	stopAndLog := func() {
+		if err := c.Stop(); err != nil {
+			log.Warnf("ChainArbitrator could not shutdown: %v", err)
+		}
+	}
+
 	// Handle all errors returned from spawning our chain watchers. If any
 	// of them failed, we will stop the chain arb to shutdown any active
 	// goroutines.
 	for err := range watcherErrs {
 		if err != nil {
-			c.Stop()
+			stopAndLog()
 			return err
 		}
 	}
 
+	// Before we start all of our arbitrators, we do a preliminary state
+	// lookup so that we can combine all of these lookups in a single db
+	// transaction.
+	startStates := make(
+		map[wire.OutPoint]*chanArbStartState, len(c.activeChannels),
+	)
+
+	err = kvdb.View(c.chanSource, func(tx walletdb.ReadTx) error {
+		for _, arbitrator := range c.activeChannels {
+			startState, err := arbitrator.getStartState(tx)
```

Comment:
Could we fetch this state a bit earlier, and pass it as part of `NewChannelArbitrator`?

Reply:
This is a bit messy to do because the lookup needs the `arbLog` which needs the `arbCfg`, so we'd need to create all configs/logs, then do the lookups, then go through all our channels again and create the arbs. I think it's cleaner left as is?